### PR TITLE
feat: add new fullscreen map page

### PR DESCRIPTION
**Pull request by ErickSharp**
Add a fullscreen map page that allows you to scroll unlike the one built into the homepage. 

Desktop:
image

Mobile:
image

### CODE REVIEW

**Comment by ErickSharp** [comment]
Note that there is currently nothing within the website such as a button or link that directs to this /map page. I may look into adding this, but I just did this PR at the request of someone and I don't know if it's worth doing so.

File: src/components/map/MapComp.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { Map } from '@flybywiresim/map';
+
+export const MapComp: React.FC = () => {
+    return (
+        <>
+            <div className=" h-screen bg-blue-dark-contrast">
+                <div className="relative h-screen pt-20 text-black z-20">
+                    <Map refreshInterval={10_000} disableMenu={true} disableWeather={false} disableScroll={false} forceTileset={'carto-light'} />
+                </div>
+            </div>
```
Comment (nistei) [inline src/components/map/MapComp.tsx:11]: The map has most of its features disabled on the home page. By removing all properties these features get unlocked which we should definitely do on the fullscreen version.

```suggestion
            <div className=" h-screen bg-blue-dark-contrast">
                <div className="relative h-screen pt-20 text-black z-20">
                    <Map />
                </div>
            </div>
```

Please also take a look at the community section. @Benjozork implemented logic to throttle updates when the map isn't in the foreground. This throttling should be used here as well.

**Comment by ErickSharp** [comment]
Buen idea.

**Comment by ErickSharp** [comment]
Penny for your thoughts: Should we perhaps include a link at the bottom of the page to access the /map extension? In addition, should I remove this component from the dedicated "map" folder and move it to say utils, or is it fine in its current location? Let me know what you guys think.

**Comment by nistei** [comment]
> Penny for your thoughts: Should we perhaps include a link at the bottom of the page to access the /map extension? In addition, should I remove this component from the dedicated "map" folder and move it to say utils, or is it fine in its current location? Let me know what you guys think.

I think we should even add it to the NavBar. The component should stay in the map directory imo. Please add the update interval logic from the [community](https://github.com/flybywiresim/website/blob/staging/src/components/home/Community.tsx#L118) section to reduce polling.

**Comment by ErickSharp** [comment]
Is there a way to at first serve them the map with the white theme so they are not confused to all hell as to why they are being served this random-ass black map instead of the white one they are used to?

**Comment by NathanInnes** [comment]
> Is there a way to at first serve them the map with the white theme so they are not confused to all hell as to why they are being served this random-ass black map instead of the white one they are used to?

Yes by adding `<Map forceTileset={'carto-light'} />` you can also try 'os-map' instead to see what suits best.

**Comment by nistei** [comment]
> Yes by adding `<Map forceTileset={'carto-light'} />` you can also try 'os-map' instead to see what suits best.

No, this will enforce it. It isn't changeable at all when setting this.

**Comment by NathanInnes** [comment]
Doesn't the dropdown menu on the map have options?

**Comment by ErickSharp** [comment]
> > Yes by adding `<Map forceTileset={'carto-light'} />` you can also try 'os-map' instead to see what suits best.
> 
> No, this will enforce it. It isn't changeable at all when setting this.

The website seems to remember your preference for the color of the map, it just defaults to the black theme, do you know of any attribute that could override the default for black?

**Comment by ErickSharp** [comment]
> Doesn't the dropdown menu on the map have options?

Yep, it would be nice to have to greet the user with the map color that they see on the homepage, however.

File: src/components/map/MapComp.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React, { useEffect, useState } from 'react';
+import { Map } from '@flybywiresim/map';
+
+export const MapComp: React.FC = () => {
```
Comment (Benjozork) [inline src/components/map/MapComp.tsx:4]: Just call this `FullscreenMap`

Comment (ErickSharp) [reply src/components/map/MapComp.tsx]: Good call

**Comment by ErickSharp** [comment]
Wouldn't say that this is a priority. I believe the map just picks the theme at the top of the list and in this case that just so happens to be the dark theme. Most plausible solution would be just to make the white theme the one at the top so that's what's picked first.

**Comment by nistei** [comment]
> Wouldn't say that this is a priority. I believe the map just picks the theme at the top of the list and in this case that just so happens to be the dark theme. Most plausible solution would be just to make the white theme the one at the top so that's what's picked first.

That's correct. The default is the first on the list and it remembers the selected one. This is best changed once we've moved the map to our component library, which is in progress.

**Review by nistei — APPROVED** [review]

**Review by NathanInnes — APPROVED** [review]

**Pull request merged**